### PR TITLE
Add VGG and pretained models

### PR DESCRIPTION
**Pull request by yl-1993**

### CODE REVIEW

File: mmcls/models/backbones/vgg.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    """3x3 convolution with padding."""
+    return nn.Conv2d(
+        in_planes,
+        out_planes,
+        kernel_size=3,
+        padding=dilation,
+        dilation=dilation)
+
+
+def make_vgg_layer(inplanes,
```
Comment (xiaojieli0903) [inline mmcls/models/backbones/vgg.py:19]: better to change `inplanes` to `in_channels` and change `planes` to `out_channels` following the naming strategies of other backbones.

Comment (xiaojieli0903) [inline mmcls/models/backbones/vgg.py:19]: better to change `inplanes` to `in_channels` and change `planes` to `out_channels` following the naming strategies of other backbones.

File: mmcls/models/backbones/vgg.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+    def __init__(self,
+                 depth,
+                 with_norm=False,
+                 num_classes=-1,
+                 num_stages=5,
+                 dilations=(1, 1, 1, 1, 1),
+                 out_indices=None,
+                 frozen_stages=-1,
+                 conv_cfg=None,
```
Comment (xiaojieli0903) [inline mmcls/models/backbones/vgg.py:80]: Please add docstring for `conv_cfg`. By the way, it is not used in other places.

Comment (xiaojieli0903) [inline mmcls/models/backbones/vgg.py:80]: Please add docstring for `conv_cfg`. By the way, it is not used in other places.

Comment (xiaojieli0903) [inline mmcls/models/backbones/vgg.py:80]: Please add docstring for `conv_cfg`. By the way, it is not used in other places.

File: mmcls/models/backbones/vgg.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        super(VGG, self).init_weights(pretrained)
+        if pretrained is None:
+            for m in self.modules():
+                if isinstance(m, nn.Conv2d):
+                    kaiming_init(m)
+                elif isinstance(m, _BatchNorm):
+                    constant_init(m, 1)
+                elif isinstance(m, nn.Linear):
+                    normal_init(m, std=0.01)
+        else:
```
Comment (xiaojieli0903) [inline mmcls/models/backbones/vgg.py:148]: There is no implementation when `pretrained` is str.

Comment (yl-1993) [reply mmcls/models/backbones/vgg.py:148]: When `pretrained` is str, it is handled in [base_backbone]( https://github.com/open-mmlab/mmclassification/blob/master/mmcls/models/backbones/base_backbone.py#L29).

**Comment by xiaojieli0903** [comment]
LGTM

File: mmcls/models/backbones/vgg.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        if self.num_classes > 0:
+            self.classifier = nn.Sequential(
+                nn.Linear(512 * 7 * 7, 4096),
+                nn.ReLU(True),
+                nn.Dropout(),
+                nn.Linear(4096, 4096),
+                nn.ReLU(True),
+                nn.Dropout(),
+                nn.Linear(4096, num_classes),
+            )
```
Comment (hellock) [inline mmcls/models/backbones/vgg.py:145]: Should this be part of head?

Comment (yl-1993) [reply mmcls/models/backbones/vgg.py:145]: VGG's 3-layer MLP classifier is sort of a classical design and thus we leave this original design inside the backbone. If users want to customize the classifier, they can set `num_classes=0` and specify their own head in config file.

**Review by hellock — APPROVED** [review]

**Pull request merged**